src/common/snippets/src/generator.cpp

```diff
@@ -3,224 +3,102 @@
-                }
-            }
-        }
-        updated_tile.push_back(op);
-    }
-
-    tail = std::move(updated_tile);
-}
-
-ngraph::snippets::code ngraph::snippets::Generator::generate(std::shared_ptr<ov::Model>& m,
-                                                             const GeneratorConfig& config,
-                                                             const void* compile_params) {
+Generator::LoweringResult Generator::generate(std::shared_ptr<ov::Model>& m, const lowered::Config& config, const void* compile_params) {
     OV_ITT_SCOPED_TASK(ngraph::pass::itt::domains::SnippetsTransform, "Snippets::Generator::generate")
+    OV_ITT_TASK_CHAIN(GENERATE, ngraph::pass::itt::domains::SnippetsTransform, "Snippets::Generator", "::Transformations")
     if (!target->is_supported())
         OPENVINO_THROW("unsupported architecture for code generation");
 
-    OV_ITT_TASK_CHAIN(GENERATE, ngraph::pass::itt::domains::SnippetsTransform, "Snippets::Generator", "::VectorTile")
-    // vector loop
-    std::vector<AllocatedEmitter> lowered;
-    auto lower_ops = [&lowered, this](const NodeVector& ops){
-        std::transform(ops.begin(), ops.end(), std::back_inserter(lowered),
-                       [this](const std::shared_ptr<Node>& n){
-                           return std::make_pair(target->get(n->get_type_info())(n), ngraph::snippets::getRegisters(n));
-                       });
+    auto linear_ir = lowered::LinearIR(m, config);
+    const size_t vector_size = get_target_machine()->get_lanes();
+    const int32_t buffer_allocation_rank = static_cast<int32_t>(config.m_loop_depth);
+
+    // Note: The pass InitLoops uses LoopInfo that contains entry and exit points of the corresponding Loop.
+    //       To avoid the Loop information corruption, we should call the passes with Load/Store work
+    //       (for example, LoadMoveBroadcastToBroadcastLoad()) after explicit Loop insertion (InitLoops())
+    lowered::pass::TransformationPipeline common_pipeline;
+    common_pipeline.register_transformation<lowered::pass::MarkLoops>(vector_size);
+    common_pipeline.register_transformation<lowered::pass::SoftmaxDecomposition>(vector_size);
+    common_pipeline.register_transformation<lowered::pass::FuseLoops>();
+    common_pipeline.register_transformation<lowered::pass::MoveResultOutOfLoop>();
+    common_pipeline.register_transformation<lowered::pass::InsertBuffers>(buffer_allocation_rank);
+    common_pipeline.register_transformation<lowered::pass::InsertLoadStore>(vector_size);
+    common_pipeline.register_transformation<lowered::pass::SetScalarCountForLoadStore>();
+    common_pipeline.register_transformation<lowered::pass::InitLoops>();
+    common_pipeline.register_transformation<lowered::pass::MoveScalarToConsumer>();
+    common_pipeline.register_transformation<lowered::pass::LoadMoveBroadcastToBroadcastLoad>();
+    common_pipeline.register_transformation<lowered::pass::PropagateLayout>();  // or should be in final?
```

Comment:
Yes, I think it's better to place it in final, since it's essentially a cleanup pass.